Ls controller spec

landrysoules opened the pull request:
Had a lot of problems here....
Started with mocking everything in order to test controller's behaviour. But then I realized that controller essentially works with ActiveRecord and then it's useless to test functions like save, update etc on mocked objects.
Therefore I wrote my spec from scratch, thinking that it's only worth testing that for every action, the controller assigns the right variable, and renders the right template, but I'm also in trouble: my tests won't pass !

landrysoules committed:
Snippets controller specs.

landrysoules committed:
Explication commit...

I had really hard times to figure out how to make relevant controller
tests with mocks/stubs everywhere. This commit is only here to discuss
my silly attempts with Aleksandar and Igor.
The second part represents my 1st attempts (mocks everywhere), while the
first part is what I started to rewrite.

landrysoules committed:
Snippet controller spec from scratch...

I'm starting to write the spec from scratch, in a way I think is more
relevant...

shiroyasha commented:
It's understandable. I also had a lot of trouble with controller specs when I started out with Rails.

Let's talk about them live :)

landrysoules commented:
OK, see you this afternoon :-)

landrysoules committed:
Snippets controller spec rewritten.

Only controller methods are now tested: external objects/methods are
mocked/stubbed.
